src/parachain/rewards.ts

```diff
@@ -209,6 +242,23 @@ export class DefaultRewardsAPI implements RewardsAPI {
         return decodeFixedPointType(rawStake);
     }
 
+    async getStakingPoolTotalStake(
+        collateralCurrencyIdLiteral: CollateralIdLiteral,
+        vaultAccountId: AccountId,
+        nonce?: number
+    ): Promise<Big> {
+        if (nonce === undefined) {
+            nonce = await this.getStakingPoolNonce(collateralCurrencyIdLiteral, vaultAccountId);
+        }
+        const collateralCurrency = currencyIdToMonetaryCurrency(
+            newCurrencyId(this.api, collateralCurrencyIdLiteral)
+        ) as CollateralCurrency;
+        const vaultId = newVaultId(this.api, vaultAccountId.toString(), collateralCurrency, this.wrappedCurrency);
+        const head = await this.api.rpc.chain.getFinalizedHead();
```

Comment:
Please don't use the finalized head approach. Only use the finalized head on cross-chain interactions (i.e., where a BTC or other cross-chain transaction is required). Otherwise, directly query the endpoint.